xarray/core/common.py

```diff
@@ -1510,7 +1510,7 @@ def _full_like_variable(other, fill_value, dtype: DTypeLike = None):
     if fill_value is dtypes.NA:
         fill_value = dtypes.get_fill_value(dtype if dtype is not None else other.dtype)
 
-    if isinstance(other.data, dask_array_type):
+    if is_duck_dask_array(other.data):
```

Comment:
does `dask.array.Array` define `numpy.full_like`? If it does, we could just remove this code path (doesn't have to be this PR, though).

Reply:
then this also depends on `numpy>=1.16`